minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/command/StartRunner.java

```diff
@@ -279,15 +280,33 @@ private void regenerateFlowConfiguration(String flowConfigFileLocation) throws C
         }
     }
 
-    private Process startMiNiFi() throws IOException {
+    private Process startMiNiFi(BootstrapProperties bootstrapProperties) throws IOException {
+        int bootstrapListenPort = parseBootstrapListenPort(bootstrapProperties);
+
         MiNiFiListener listener = new MiNiFiListener();
-        listenPort = listener.start(runMiNiFi, bootstrapFileProvider, configurationChangeListener);
+        listenPort = listener.start(runMiNiFi, bootstrapListenPort, bootstrapFileProvider, configurationChangeListener);
 
         CMD_LOGGER.info("Starting Apache MiNiFi...");
 
         return startMiNiFiProcess(getProcessBuilder());
     }
 
+    private int parseBootstrapListenPort(BootstrapProperties bootstrapProperties) {
+        String bootstrapListenPort = bootstrapProperties.getProperty(NIFI_BOOTSTRAP_LISTEN_PORT);
+        if (bootstrapListenPort != null) {
+            try {
+                int port = Integer.parseInt(bootstrapListenPort.trim());
+                CMD_LOGGER.info("Using boostrap listen port {}", port);
+                return port;
+            } catch (Exception e) {
+                CMD_LOGGER.warn("Unable to parse bootstrap listen port, falling back to default (random) port");
+                return 0;
```

Comment:
Instead of logging a warning, it seems like this should throw an `IllegalArgumentException`. The configuration attempted to specify a port number, so starting on a random port is contrary to the configured behavior.